src/sage/rings/polynomial/integer_valued_polynomials.py

```diff
@@ -832,18 +831,48 @@ def h_polynomial(self):
                 """
                 Return the `h`-vector as a polynomial.
 
-                .. SEEALSO:: :meth:`h_vector`
+                .. SEEALSO:: :meth:`h_vector`, :meth:`fraction`
 
                 EXAMPLES::
 
                     sage: x = polygen(QQ,'x')
                     sage: A = IntegerValuedPolynomialRing(ZZ).S()
                     sage: ex = A.from_polynomial((1+x)**3)
                     sage: ex.h_polynomial()
-                    z^3 + 4*z^2 + z
+                    z^2 + 4*z + 1
                 """
                 anneau = PolynomialRing(self.parent().base_ring(), 'z')
-                return anneau(list(self.h_vector()))
+                return anneau(list(reversed(self.h_vector())))
+
+            def fraction(self):
+                """
+                Return the generating series of values as a fraction.
+
+                In the case of Ehrhart polynomials, this is known as
+                the Ehrhart series.
+
+                .. SEEALSO:: :meth:`h_vector`, :meth:`h_polynomial`
+
+                EXAMPLES::
+
+                    sage: A = IntegerValuedPolynomialRing(ZZ).S()
+                    sage: ex = A.monomial(4)
+                    sage: f = ex.fraction();f
+                    -1/(t^5 - 5*t^4 + 10*t^3 - 10*t^2 + 5*t - 1)
```

Comment:
Can we also add a test where the numerator is not a constant?

Can you check the corner case for the h-vector being empty for 0 and returning an element in the same parent?